Fortnox.NET.Tests/Fortnox.NET.Tests/WebSocket/FortnoxWebSocketClientTest.cs

```diff
@@ -21,237 +22,105 @@ public class FortnoxWebSocketClientTest : TestBase
-                            }
-                        }
-                    }).GetAwaiter().GetResult();
-                }
-                catch (Exception e)
-                {
-                    Assert.Fail("");
-                }
-            }, TaskCreationOptions.LongRunning);
+                response = await client.Recieve();
+            }
 
-            task.RunSynchronously();
+            Assert.IsTrue(response.Topic == "articles");
+            Assert.IsTrue(response.EventType == "article-updated-v1");
+            Assert.IsTrue(response.EntityId == "100370");   
         }
 
         [TestMethod]
         public async Task CannotConnectTwice()
         {
-            var client = new FortnoxWebSocketClient(this.connectionSettings.AccessToken, this.connectionSettings.ClientSecret)
-                .AddTopic(WebSocketTopic.Articles);
-
+            var client = new FortnoxWebSocketClient(this.connectionSettings.ClientSecret);
             await client.Connect();
             await Assert.ThrowsExceptionAsync<InvalidOperationException>(() => client.Connect());
         }
 
-        [TestMethod]
-        [ExpectedException(typeof(WebSocketException))]
-        public async Task UserHasToSpecifyTopicsOrThrowsException()
-        {
-            var client = await new FortnoxWebSocketClient(this.connectionSettings.AccessToken, this.connectionSettings.ClientSecret)
-                .Connect();
-        }
-
-        [TestMethod]
-        [ExpectedException(typeof(WebSocketException))]
-        public async Task InvalidCredentialsThrowsException()
-        {
-            var client = await new FortnoxWebSocketClient("NotAnAccessToken", "NotASecret")
-                .Connect();
-        }
-
         [TestMethod]
         public async Task CanSubscribeToMultipleTopics()
         {
             var listeningTask = new Task(async () =>
             {
-                var client = new FortnoxWebSocketClient(this.connectionSettings.AccessToken, this.connectionSettings.ClientSecret)
-                    .AddTopic(WebSocketTopic.Articles)
-                    .AddTopic(WebSocketTopic.Orders);
+                var client = new FortnoxWebSocketClient(this.connectionSettings.ClientSecret);
+                await client.Connect();
+                await client.AddTenant(this.connectionSettings.AccessToken);
+                await client.AddTopic(WebSocketTopic.Articles);
+                await client.AddTopic(WebSocketTopic.Orders);
+                await client.Subscribe();
 
-                await client.ConnectAndListen(async (socket) =>
-                {
-                    var recievedArticleUpdate = false;
-                    var recievedOrderUpdate = false;
+                var recievedArticleUpdate = false;
+                var recievedOrderUpdate = false;
 
-                    foreach (var response in client.GetNextEvent(socket))
+                while (!recievedArticleUpdate && !recievedOrderUpdate)
```

Comment:
if this test for some weird reason failed to receive the expected messages, will it run forever?

Reply:
would this be a good case to add tests for cancelation tokens?

Reply:
I believe that adding a timeout in the form of cancellation tokens sets the internal state of the underlying ClientWebSocket to "aborted" which makes the Client unusable afterwards. I will add an additional test to confirm as well as support for Cancellation tokens.

Reply:
great!

Reply:
A shame that the socket cannot be re-used after the cancellation though. Does it get reset if the message is received within the timout limit?

Reply:
That is correct.

The following code snippet would not abort the underlying socket:
```CSharp
CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
await client.Recieve(cancellationTokenSource.Token);
cancellationTokenSource.Cancel();
```